### PR TITLE
bake: ulimits and shm-size support

### LINKED ISSUE #2241: buildx bake file doesn't support new ulimit directives
### Description

Not sure if this is a bug, or just feature oversight.

https://github.com/compose-spec/compose-spec/pull/433 implemented support for ulimit directives in Docker Compose file specification, however if I try to use new directives in BuildX Bake file it produces an error.

```
services:
  mydefaultbuild:
    build:
      ulimits:
        nofile:
          soft: 65536
          hard: 65536
```

```
[] docker buildx bake
[+] Building 0.0s (1/1) FINISHED                                                                         docker:default
 => [internal] load local bake definitions                                                                         0.0s
 => => reading docker-compose.yaml 139B / 139B                                                                     0.0s
ERROR: validating : services.mydefaultbuild.build Additional property ulimits is not allowed
```

### PR DESCRIPTION
**Pull request by crazy-max**
fixes #2241

### CODE REVIEW

File: docs/bake-reference.md [hunk 1]
```diff
@@ -832,6 +834,20 @@ RUN --mount=type=secret,id=KUBECONFIG \
     KUBECONFIG=$(cat /run/secrets/KUBECONFIG) helm upgrade --install
 ```
 
+### `target.shm-size`
```
Comment (tonistiigi) [inline docs/bake-reference.md:837]: The docs should describe a bit about the feature as well, so what it does and how it applies to build containers, not only the formatting of the value. Same for ulimits that can define the values.

Maybe it also makes sense to mention that this is advanced customization that usually is not needed and should be let for builder to figure out automatically.

**Review by dvdksn — COMMENTED** [review]
Left some suggestions / some I am not sure about so feel free to dismiss if you disagree

File: docs/bake-reference.md [hunk 2]
```diff
@@ -832,6 +834,28 @@ RUN --mount=type=secret,id=KUBECONFIG \
...
+
+Sets the size of the shared memory allocated for build's containers when using
+`RUN` instructions.
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply docs/bake-reference.md]: I don't think it makes sense at buildx level imo

**Review by dvdksn — APPROVED** [review]
LGTM

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**